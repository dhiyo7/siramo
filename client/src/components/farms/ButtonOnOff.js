import React, { Component } from 'react'
import {
  View,
  Text,
  TouchableOpacity,
  StyleSheet
} from 'react-native'
import { inject, observer } from 'mobx-react'
import UserStore from '../../store/UserStore'

@inject('UserStore')
class ButtonOnOff extends Component {
  render() {
    const {ready_siram, triggerSiram} = this.props.siram
    let txtButton = 'Start Watering'
    let buttonStyles = styles.buttonContainerStart
    if (ready_siram === 0) {
      txtButton = 'Stop Watering'
      buttonStyles = styles.buttonContainerStop
    }
    return (
      <TouchableOpacity
        style={buttonStyles}
        onPress={triggerSiram}
      >
        <Text style={styles.buttonText}>{txtButton}</Text>
      </TouchableOpacity>
    )
  }
}

const styles = StyleSheet.create({
  buttonContainerStart: {
    backgroundColor: '#33691E',
    paddingVertical: 15,
    marginBottom: 4,
    borderRadius: 4
  },
  buttonContainerStop: {
    backgroundColor: '#f44242',
    paddingVertical: 15,
    marginBottom: 4,
    borderRadius: 4
  },
  buttonText: {
    color: '#fff',
    textAlign: 'center',
    fontWeight: '700'
  }
})

export default ButtonOnOff