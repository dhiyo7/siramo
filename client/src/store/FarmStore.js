import { observable, computed } from 'mobx'
import { db, User } from './firebase'
import { Alert } from 'react-native'


class FarmStore {
  @observable farmData = {
    historyHumidity: [],
    historyWaterRatio: [],
    historyTemperature: [],
    historyWaterLevel: [],
    historyLoading: false
  }

  @observable navigation = {}

  getHistory = (key) => {
    this.farmData.historyLoading = true
    db.ref('/history/17UFak7JqufG1RXUeVW30jwdfrQ2').on('value', (snapshot) => {

      snapshot.forEach(snap => {
        this.farmData.historyHumidity.push({
          x: snap.val().last_updated,
          y: parseFloat(snap.val().humidity)
        })
        this.farmData.historyWaterRatio.push({
          x: snap.val().last_updated,
          y: Math.round(parseFloat(snap.val().water_ratio))
        })
        this.farmData.historyTemperature.push({
          x: snap.val().last_updated,
          y: parseFloat(snap.val().temperature)
        })
        this.farmData.historyWaterLevel.push({
          x: snap.val().last_updated,
          y: parseFloat(snap.val().water_level)
        })
      })
      this.farmData.historyLoading = false
    })
  }

  setSchedule = (cronFormat, max, min) => {
    let updatedValue = {
      name: cronFormat,
      maxWaterRatio: max,
      minWaterRatio: min
    }
    console.log('Firebase Updated', updatedValue)
    db.ref(`/users`).update(updatedValue)
      .then(response => {
        console.log(response)
      })
      .catch(err => {
        console.log(err)
      })
  }

}

export default new FarmStore()