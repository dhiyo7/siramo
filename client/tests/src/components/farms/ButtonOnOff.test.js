import React from 'react'
import {
  View,
  Text,
  TouchableOpacity,
  StyleSheet
} from 'react-native'
import Enzyme, { mount, shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import renderer from 'react-test-renderer'
import ButtonOnOff from '../../../../src/components/farms/ButtonOnOff'
import { Ionicons } from '@expo/vector-icons'
import FarmStore from '../../../../src/store/FarmStore'

Enzyme.configure({ adapter: new Adapter()})

let wrapper
let propSiram 

describe('<ButtonOnOff /> snapshot Testing', () => {
  it('snapshot works!', () => {
    propSiram = {ready_siram: 0, triggerSiram: FarmStore.triggerSiram}
    const tree = renderer.create(<ButtonOnOff siram={propSiram}/>)
    expect(tree).toMatchSnapshot()
  })
})

describe('<ButtonOnOff /> Text and Color Test', () => {
  
  it('should changed according to ready_siram status', () => {
    propSiram = {ready_siram: 0, triggerSiram: FarmStore.triggerSiram}
    wrapper = shallow(<ButtonOnOff siram={propSiram}/>)
    expect(wrapper.find('TouchableOpacity').get(0).props.style.backgroundColor).toEqual('#31ce12')
    expect(wrapper.find('Text').get(0).props.children[2]).toEqual('Start Watering')

    propSiram = {ready_siram: 1, triggerSiram: FarmStore.triggerSiram}
    let wrapperStop = shallow(<ButtonOnOff siram={propSiram}/>)
    expect(wrapperStop.find('TouchableOpacity').get(0).props.style.backgroundColor).toEqual('#f44242')
    expect(wrapperStop.find('Text').get(0).props.children[2]).toEqual('Stop Watering')
  })
})