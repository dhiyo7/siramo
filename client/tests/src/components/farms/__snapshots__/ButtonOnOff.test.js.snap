// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ButtonOnOff /> snapshot Testing snapshot works! 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  collapsable={undefined}
  hasTVPreferredFocus={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "#31ce12",
      "borderRadius": 4,
      "marginBottom": 4,
      "marginTop": 8,
      "opacity": 1,
      "paddingVertical": 15,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#fff",
        "fontSize": 20,
        "fontWeight": "700",
        "textAlign": "center",
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
    />
     
    Start Watering
  </Text>
</View>
`;
